torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -1151,15 +1151,10 @@ def _finalize_params(
     if not handle:
         return
     flat_param = handle.flat_param
-    if hasattr(flat_param, "_post_backward_hook_state"):
-        post_backward_hook_state_len = len(flat_param._post_backward_hook_state)
-        expected_post_backward_hook_state_len = int(flat_param.requires_grad) + 1
-        _p_assert(
-            post_backward_hook_state_len == expected_post_backward_hook_state_len,
-            f"Invalid: ``_post_backward_hook_state``: {flat_param._post_backward_hook_state}",
-        )
-        flat_param._post_backward_hook_state[-1].remove()
-        delattr(flat_param, "_post_backward_hook_state")
+    if hasattr(flat_param, "_post_backward_hook_handle"):
+        pbhs_handle = flat_param._post_backward_hook_handle
+        pbhs_handle.remove()
+        delattr(flat_param, "_post_backward_hook_handle")
```

Comment:
Very much style preference, feel free to ignore
```suggestion
        flat_param._post_backward_hook_handle.remove()
        del flat_param._post_backward_hook_handle
```